examples/ephemeral-spark/tamr-config.tf

```
@@ -24,39 +25,46 @@ module "tamr-config" {
   spark_cluster_log_uri          = "s3n://${module.s3-logs.bucket_name}/${var.path_to_spark_logs}"
+  spark_driver_memory            = "5G"
+  spark_executor_instances       = 15
+  spark_executor_memory          = "6G"
+  spark_executor_cores           = 1
   tamr_data_path                 = "tamr/unify-data"
   tamr_spark_config_override     = "[{'name' : 'sparkOverride1','executorInstances' : '2','sparkProps' : {'spark.cores.max' : '4'}},{'name' : 'sparkOverride2','driverMemory' : '4G','executorMemory' : '5G'}]"
   tamr_spark_properties_override = "{'spark.driver.maxResultSize':'4g'}"
 
-  # ElasticSearch
-  es_domain_endpoint = module.tamr-es-cluster.tamr_es_domain_endpoint
-
-  # ESP
-  tamr_external_storage_providers = "[{'name' : 's3a_tamr_config_test','description' : 'The S3a filesystem at root of ${module.s3-data.bucket_name}','uri' : 's3a://${module.s3-data.bucket_name}/'}]"
-
   # Ephemeral Spark
-  emr_release_label           = "emr-5.29.0" # spark 2.4.4
-  emr_instance_profile_name   = module.ephemeral-spark-iam.emr_ec2_instance_profile_name
-  emr_service_role_name       = module.ephemeral-spark-iam.emr_service_role_name
-  emr_key_pair_name           = module.emr_key_pair.key_pair_key_name
-  emr_subnet_id               = var.ec2_subnet_id
-  master_instance_type        = "m4.large"
-  master_ebs_volumes_count    = 1
-  master_ebs_size             = 50
-  master_ebs_type             = "gp2"
-  core_ebs_volumes_count      = 1
-  core_ebs_size               = 200
-  core_ebs_type               = "gp2"
-  core_group_instance_count   = 4
-  core_instance_type          = "r5.4xlarge"
-  emr_service_access_sg_id    = module.ephemeral-spark-sgs.emr_service_access_sg_id
-  emr_managed_master_sg_id    = module.ephemeral-spark-sgs.emr_managed_master_sg_id
-  emr_additional_master_sg_id = module.ephemeral-spark-sgs.emr_additional_master_sg_id
-  emr_managed_core_sg_id      = module.ephemeral-spark-sgs.emr_managed_core_sg_id
-  emr_additional_core_sg_id   = module.ephemeral-spark-sgs.emr_additional_core_sg_id
+  emr_release_label         = "emr-5.29.0" # spark 2.4.4
+  emr_instance_profile_name = module.ephemeral-spark-iam.emr_ec2_instance_profile_name
+  emr_service_role_name     = module.ephemeral-spark-iam.emr_service_role_name
+  emr_key_pair_name         = module.emr_key_pair.key_pair_key_name
+  emr_subnet_id             = var.ec2_subnet_id
+  master_instance_type      = "m4.large"
+  master_ebs_volumes_count  = 1
+  master_ebs_size           = 50
+  master_ebs_type           = "gp2"
+  core_ebs_volumes_count    = 1
+  core_ebs_size             = 200
+  core_ebs_type             = "gp2"
+  core_group_instance_count = 4
+  core_instance_type        = "r5.xlarge"
+
+  emr_managed_master_sg_id = module.ephemeral-spark-sgs.emr_managed_sg_id
+  # emr_managed_master_sg_id = "" # you may leave this blank and AWS creates one automatically
+  emr_additional_master_sg_id = join(", ", module.aws-emr-sg-core.security_group_ids)
```

Not that the clusters are managed by terraform, but should we get rid of the space here too to avoid this spreading out further.

Also, I checked and our code doesn't trim whitespace values from these values, so we must be relying on AWS to do that.

```suggestion
  emr_additional_master_sg_id = join(",", module.aws-emr-sg-core.security_group_ids)
```

---

Also, I'd be in favor of making these variables lists and letting the module do this work.

---

I agree. Also, changes were done in EMR module 6.0.0 to these var names and usability inside the module. Those changes weren't reflected to tamr-config module. It was very confusing for me to set this up and it may be for future users too. For now I guess we just commit this, but I'd like to have a better revision on having an opinionated example on how to create the SGs and pass their IDs to the tamr-config (this is only needed for ephemeral-spark).